presidio-structured/presidio_structured/data/data_transformers.py

```diff
@@ -0,0 +1,212 @@
+from abc import ABC, abstractmethod
+from typing import Any, Callable, Dict, List, Union
+
+from pandas import DataFrame
+from presidio_anonymizer.entities import OperatorConfig
+from presidio_anonymizer.operators import OperatorsFactory, OperatorType
+
+from presidio_structured.config import StructuredAnalysis
+
+
+class DataTransformerBase(ABC):
```

Comment:
> Another idea: Can this / should this inherit from the `Operator` class? Or is it serving a different purpose here?

I think we can make it inherit from the Operator class since the interfaces are (somewhat) similar, but  I don't think it should. An operator is really the tool which changes some string using anonymization/decryption techniques, whereas the data_processors' purpose is to operate on a certain data structure and all the data within it. 

Perhaps it would make sense to rename the methods in the processors to avoid this confusion, but I'll leave that open for discussion for now